### DISCUSSION TITLE
Minimizer in MOOSE?

### DISCUSSION THREAD

**Discussion by heinono1** [idaholab/moose#21880] [Q&A General] [answered]
Does MOOSE have a tool or utility for direct minimization? In many problems, the static solution (or asymptotic solution) of the governing equation is an energy minimum, in particular for a large class of problems in which the energy of the system is strictly monotonically non-increasing. If one is interested in the asymptotic stationary solution, one can either time-integrate until the energy does not decrease any more, or do a direct energy minimization. Is there such a direct energy minimization path implemented in MOOSE?

**Comment by GiudGiud** [comment]
@lynnmunday might have something for you in isopod

**Comment by heinono1** [comment]
Cool - thanks! I'll look into Isopod.

Cheers,
Olle

On Thu, Aug 18, 2022 at 10:23 AM Guillaume Giudicelli <
***@***.***> wrote:

> @lynnmunday <https://github.com/lynnmunday> might have something for you
> in isopod
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/discussions/21880#discussioncomment-3424017>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEKZEF4DSIULJW6K7DKFUMLVZZIO3ANCNFSM565RXSYA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Olle Heinonen
***@***.***

**Reply by hugary1995** [reply]
Isopod essentially interfaces with PETSc/TAO. You could also use trilinos's ROL.

**Comment by GiudGiud** [comment, marked as answer]
It's coming to MOOSE
https://github.com/idaholab/moose/pull/21887

**Comment by heinono1** [comment]
Super!

On Thu, Aug 18, 2022 at 4:10 PM Guillaume Giudicelli <
***@***.***> wrote:

> It's coming to MOOSE
> #21887 <https://github.com/idaholab/moose/pull/21887>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/discussions/21880#discussioncomment-3426767>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEKZEF7R3FGN2TJJ3NOGGT3VZ2RGBANCNFSM565RXSYA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Olle Heinonen
***@***.***
